Reorganize Chainlink node documentation

dwightjl opened the pull request:
Preview URL: https://documentation-git-fork-dwightjl-main-chainlinklabs.vercel.app/chainlink-nodes

dwightjl commented:
I restored the Docker recommendation section. Previously I had moved it to the Requirements section because it seemed redundant with the Docker-CE requirement, but if you want the Docker requirement to be more prominent we can keep the full Docker section.

dwightjl commented:
These fixes are still relevant. Re-synced and ready to merge.

dwightjl commented:
Updated to track the latest changes in main. See the PR description for the current list of changes.

PatrickAlphaC commented on docs/Node Operators/running-a-chainlink-node.md:
````diff
@@ -124,49 +127,33 @@ GAS_UPDATER_ENABLED=true
 ALLOW_ORIGINS=*" > ~/.chainlink/.env
 ```
 
-### Set your Ethereum Client URL
````
This change doesn't seem clear to me. 

Do we have to do environment variables for the ETH_URL? Do we think this is better?

dwightjl replied:
> Do we have to do environment variables for the ETH_URL? Do we think this is better?

We could provide a config template and have users fill it out manually. That's what I ended up doing when I first tried this doc because the env variables didn't work as documented.

Env variables are easier for some users because copying and pasting commands is straightforward as long as the commands do exactly what we expect. My personal bias is towards manually editing a config template because that's what is fastest for me usually.

The change I made here is to fix the broken `docker inspect` command and combine URL configuration into one section with sub-options. Setting the URL is the task. Setting up the ETH client is a prerequisite at the beginning of the page.

dwightjl replied:
@PatrickAlphaC I revised this section to remove the environment variable complexity. It now asks users to create a `.env` file and edit the values. https://deploy-preview-192--dreamy-villani-0e9e5c.netlify.app/docs/running-a-chainlink-node/#create-an-environment-file

PatrickAlphaC commented:
Can we just have "running from docker" be above "running from source"? And then make it clear that the rest of the doc is going to be running from docker.

PatrickAlphaC commented:
https://deploy-preview-192--dreamy-villani-0e9e5c.netlify.app/docs/running-a-chainlink-node/#create-an-environment-file

This section is still confusing to me. 

Can we have sections for adding the 2 variables:

```
ETH_URL=ws://[CONTAINER_IP]:[CONTAINER_PORT]
DATABASE_URL=postgresql://[DB_USER]:[DB_PW]@[DB_IP]:[DB_PORT]/[DB_NAME]?sslmode=[SSL_MODE]
```

Like we had before? Like something like "now add your ETH_URL, you can do it one of 3 ways. If you're runing your own, do it like `ws://[CONTAINER_IP]:[CONTAINER_PORT]`, if you're using an external URL, do it like this: ...."

Does that make sense?

PatrickAlphaC commented:
For `Start the Chainlink Node`, we should tell them that a decent version is 0.10.10

dwightjl commented:
@PatrickAlphaC I rewrote significant parts of the document and split up the ETH client steps from the DB config steps. I also added 0.10.10 as a recommended version.

The whole document has significant changes since you last reviewed, and some additional pages are also changed.

PatrickAlphaC commented:
Sorry, going to need another day or 2 to review.

dwightjl commented:
Rebased to main

chainchad commented on src/pages/chainlink-nodes/configuring-nodes.mdx:
````diff
@@ -0,0 +1,127 @@
+   docker exec -it chainlink cat /home/chainlink/config.toml > ./config.toml
+   ```
+
+1. Create a `secrets.toml` file with the minimum required secrets, which are `[Database]` and `[Password]`. If you are working on a test node, you can use `?sslmode=disable` in the database `URL`:
````
Should we also add a note about adding `[Explorer]` and any other relevant sections from the [SECRETS.md](https://github.com/smartcontractkit/chainlink/blob/v1.13.0/docs/SECRETS.md#explorer) doc? I see the doc is mentioned below, but I wonder if it'd be good to explicitly call this one out.

dwightjl replied:
Added. Does the description make sense?

chainchad replied:
Perfect, thanks!

chainchad reviewed (approved):
👍

krehermann commented on src/pages/chainlink-nodes/configuring-nodes.mdx:
````diff
@@ -0,0 +1,127 @@
+   docker run --name chainlink-config-validator -v ~/.chainlink-toml:/chainlink -it --rm smartcontract/chainlink:1.13.0 -config /chainlink/config.toml -secrets /chainlink/secrets.toml config validate
+   ```
+
+   You will likely see some invalid config errors. For example:
````
btw, the 'fix' for this is add to secrets.toml:
`AllowSimplePasswords = true`

dwightjl replied:
Really the node op should go into Postgres and make the password sufficiently complex, but I added a sentence linking to `AllowSimplePasswords` so people can more quickly get their nodes running.

krehermann commented on src/pages/chainlink-nodes/configuring-nodes.mdx:
````diff
@@ -0,0 +1,127 @@
+   docker stop chainlink
+   ```
+
+1. Make a Postgres [database snapshot](https://www.postgresql.org/docs/current/backup-file.html) so you can restore your previous Chainlink node if necessary.
````
postgres backup should link here instead https://www.postgresql.org/docs/current/backup-dump.html 

pg_dump is the preferred mechanism for generic postgres backups

dwightjl replied:
Done

krehermann commented on src/pages/chainlink-nodes/configuring-nodes.mdx:
````diff
@@ -0,0 +1,127 @@
+1. Start a new Chainlink node Docker container named using the new `config.toml` and `secrets.toml` files. This example uses `chainlink-toml` as the container name:
+
+   ```shell
+   docker run --platform linux/x86_64/v8 --name chainlink-toml  -v ~/.chainlink-toml:/chainlink -it -p 6688:6688 --add-host=host.docker.internal:host-gateway smartcontract/chainlink:1.13.0 local n -config /chainlink/config.toml -secrets /chainlink/secrets.toml
````
coming from the eng side, i'd strongly prefer `node start` to `local n` the former convey more intention and better represents what actually happens 


docker run --platform linux/x86_64/v8 --name chainlink-toml -v ~/.chainlink-toml:/chainlink -it -p 6688:6688 --add-host=host.docker.internal:host-gateway smartcontract/chainlink:1.13.0 node start -config /chainlink/config.toml -secrets /chainlink/secrets.toml

krehermann replied:
also, the command didn't work for me on 1.13.0 i think the config and secrets flags have to be at the top level, unfortunately. this changes for 2.0.0, where they can be scoped for the `node` command

1.13:
```
docker run --platform linux/x86_64/v8 --name chainlink-toml -v ~/.chainlink-goerli:/chainlink -it -p 6688:6688 --add-host=host.docker.internal:host-gateway smartcontract/chainlink:1.13.0  -config /chainlink/config.toml -secrets /chainlink/secrets.toml node start
```

2.0.0:
the above, or, PREFERED:

```
docker run --platform linux/x86_64/v8 --name chainlink-toml -v ~/.chainlink-goerli:/chainlink -it -p 6688:6688 --add-host=host.docker.internal:host-gateway smartcontract/chainlink:1.13.0  node -config /chainlink/config.toml -secrets /chainlink/secrets.toml  start
```

dwightjl replied:
Updated to `node start` and corrected the flag order.

krehermann commented on src/pages/chainlink-nodes/configuring-nodes.mdx:
````diff
@@ -0,0 +1,127 @@
+To specify multiple config files, add additional `-config` flags to the `docker run` command:
+
+```shell
+docker run --platform linux/x86_64/v8 --name chainlink-toml  -v ~/.chainlink-toml:/chainlink -it -p 6688:6688 --add-host=host.docker.internal:host-gateway smartcontract/chainlink:1.13.0 local n -config /chainlink/config.toml -config /chainlink/config2.toml -config /chainlink/config3.toml -secrets /chainlink/secrets.toml
````
placement of `-config, -secret` as above

dwightjl replied:
Done

jmank88 commented on src/pages/chainlink-nodes/configuring-nodes.mdx:
````diff
@@ -0,0 +1,127 @@
+   ```
+
+1. Edit the `config.toml` file and manually correct any errors. See the [Node Config](/chainlink-nodes/v1/node-config) and [Node Secrets](/chainlink-nodes/v1/secrets-config) pages to learn which config settings are valid. For the error in this example, an EVM chain was configured with no nodes. Removing this from the config made the config valid:
+
````
The examples at the top of the config and secrets docs are meant to be minimalish. There are only five total required fields (for evm anyways), although `WSURL` is also conditionally required since with one node it can't be send-only.
https://github.com/smartcontractkit/chainlink/blob/develop/docs/CONFIG.md#example
https://github.com/smartcontractkit/chainlink/blob/develop/docs/SECRETS.md#example

jmank88 replied:
> is there are a way to generate a TOML dump will all the defaults omitted, so that you can see which fields have been explicitly configured

What do you mean? This is how the literal `dump` sub-command works, and the logs on startup distinguish input from effective (w/ defaults).

dwightjl replied:
The [Running a Chainlink Node](https://documentation-git-fork-dwightjl-main-chainlinklabs.vercel.app/chainlink-nodes/v1/running-a-chainlink-node#run-chainlink-node) guide has a minimal config to help users get started. Does that address your request here?

krehermann replied:
>What do you mean? This is how the literal dump sub-command works, and the logs on startup distinguish input from effective (w/ defaults).

in this doc, the user is instructed the run a `chainlink-validator` docker image. it prints out the entire config, including defaults, which is overwhelming

krehermann replied:
> The [Running a Chainlink Node](https://documentation-git-fork-dwightjl-main-chainlinklabs.vercel.app/chainlink-nodes/v1/running-a-chainlink-node#run-chainlink-node) guide has a minimal config to help users get started. Does that address your request here?

yes, i missed the link in my first reading

jmank88 replied:
> in this doc, the user is instructed the run a chainlink-validator docker image. it prints out the entire config, including defaults, which is overwhelming

It also prints out the input separate from the full effective, just like when starting the node:
https://github.com/smartcontractkit/chainlink/blob/489280eb6530092afe8711ca059d201d7091e97e/core/services/chainlink/config_general.go#L238-L241

dwightjl replied:
PR approved from krehermann. Resolving this thread.

dwightjl commented on src/pages/chainlink-nodes/configuring-nodes.mdx:
````diff
@@ -0,0 +1,127 @@
+1. Validate the configuration:
+
+   ```shell
+   docker run --name chainlink-config-validator -v ~/.chainlink-toml:/chainlink -it --rm smartcontract/chainlink:1.13.0 -config /chainlink/config.toml -secrets /chainlink/secrets.toml config validate
````
Done

thedriftofwords commented:
Great reorg - left nav looks a lot cleaner

dwightjl commented on src/pages/chainlink-nodes/v1/running-a-chainlink-node.md:
````diff
@@ -122,25 +175,25 @@ Ganache is a mock testnet. Although you can run nodes on Ganache, it is not offi
    ```
 
-1. You can now connect to your Chainlink node's UI interface by navigating to [http://localhost:6688](http://localhost:6688). If using a VPS, you can create a [SSH tunnel](https://www.howtogeek.com/168145/how-to-use-ssh-tunneling/) to your node for `6688:localhost:6688` to enable connectivity to the GUI. Typically this is done with `ssh -i $KEY $USER@$REMOTE-IP -L 6688:localhost:6688 -N`. A SSH t ...
+1. You can now connect to your Chainlink node's UI interface by navigating to [http://localhost:6688](http://localhost:6688). If using a VPS, you can create an [SSH tunnel](https://www.howtogeek.com/168145/how-to-use-ssh-tunneling/) to your node for `6688:localhost:6688` to enable connectivity to the GUI. Typically this is done with `ssh -i $KEY $USER@$REMOTE-IP -L 6688:localhost:6688 -N`. A SSH  ...
````
This works for me.
